cal/v1/network/drivers/base.py

```diff
@@ -1,37 +1,78 @@
-import abc
-import six
-
-
-@six.add_metaclass(abc.ABCMeta)
-class BaseDriver(object):
-
-    def __init__(self, *args, **kwargs):
-        pass
-
-    @abc.abstractmethod
-    def create_network(self, name, *args, **kwargs):
-        pass
-
-    @abc.abstractmethod
-    def delete_network(self, id):
-        pass
-
-    @abc.abstractmethod
-    def list_network(self):
-        pass
-
-    @abc.abstractmethod
-    def show_network(self, id):
-        pass
-
-    @abc.abstractmethod
-    def update_network(self, id):
-        pass
-
-    @abc.abstractmethod
-    def attach_to_router(self, id):
-        pass
-
-    @abc.abstractmethod
-    def detach_to_router(self, id):
-        pass
+""" this is contain Abstract Class and Quota Class
+    for all network driver which we want to implement.
+"""
+
+
+import abc
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseDriver(object):
+    """abstract class for network driver"""
+
+    def __init__(self):
+        super(BaseDriver, self).__init__()
+        self.provider = "Unknown"
```

Comment:
Do not set the default of provider to string "Unknown". Set it to None make it easier to compare both on performance or good meaning.